libraries/chain/proposal_evaluator.cpp

```diff
@@ -56,17 +56,33 @@ namespace golos { namespace chain {
         remove_existing(required_active, required_total);
         required_total.insert(required_active.begin(), required_active.end());
 
-        // For more information, see sign_state.cpp
+        // For more information, see transaction.cpp
         FC_ASSERT(
             required_posting.empty() != required_total.empty(),
             "Can't combine operations required posting authority and active or owner authority");
         required_total.insert(required_posting.begin(), required_posting.end());
 
+        for (const auto& account: required_total) {
+            FC_ASSERT(
+                nullptr != db().find_account(account),
+                "Account '${account}' for proposed operation doesn't exist", ("account", account));
+        }
+
+        FC_ASSERT(!required_total.empty(), "No operations require approvals");
+
         transaction trx;
         for (const auto& op : o.proposed_operations) {
             trx.operations.push_back(op.op);
         }
-        trx.validate();
+        trx.set_expiration(db().head_block_time() + STEEMIT_MAX_TIME_UNTIL_EXPIRATION);
+
+        const uint32_t skip_steps =
+                db().skip_authority_check |
+                db().skip_transaction_signatures |
+                db().skip_tapos_check |
```

Comment:
Reading this code, it's not obvious, what happens inside `db()`. Rewrite this to `golos::chain::database::skip_xxx`, which clearly shows that no `db()` side-effects used.

Reply:
Fixed